spring-boot-project/spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/jar/JarFileEntries.java

```diff
@@ -188,7 +189,8 @@ public InputStream getInputStream(FileHeader entry) throws IOException {
 		}
 		InputStream inputStream = getEntryData(entry).getInputStream();
 		if (entry.getMethod() == ZipEntry.DEFLATED) {
-			inputStream = new ZipInflaterInputStream(inputStream, (int) entry.getSize());
+			inputStream = new ZipInflaterInputStream(inputStream, new Inflater(true),
```

Comment:
I don't think we want to pass in an `Inflater` and than have `ZipInflaterInputStream` call `end()` on it.